### PR TITLE
Added fallback for failed summary generation and updated results #113

### PR DESCRIPTION
**Pull request by anton164**
If a summary fails to generate, we fall back to the last summary
  generated by GEF.

Evaluation code is refactored accordingly - failed status is now read
from a dict instead of the summary text directly.

### CODE REVIEW

File: results/latex/bart-test-extrinsic-100.tex [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
+\newcommand{\bartPinocchioNonFactualExtrinsic}{22\%}
+\newcommand{\bartPinocchioNonFactualIntrinsic}{8\%}
+\newcommand{\bartPinocchioFactualExtrinsicRatio}{86\%}
 \newcommand{\bartPinocchioSkipped}{0\%}
 \newcommand{\bartMengRLFactual}{52\%}
 \newcommand{\bartMengRLNonFactual}{48\%}
 \newcommand{\bartMengRLNonFactualExtrinsic}{44\%}
 \newcommand{\bartMengRLNonFactualIntrinsic}{19\%}
-\newcommand{\bartMengRLFactualExtrinsicRatio}{74\%}
+\newcommand{\bartMengRLFactualExtrinsicRatio}{73\%}
```
Comment (anton164) [inline results/latex/bart-test-extrinsic-100.tex:35]: Unsure why this changed the Meng evaluation... need to investigate

File: tests/test_evaluation.py [hunk 2]
```diff
@@ -13,14 +13,17 @@
 def load_data(results_path: str):
```
Comment (anton164) [inline tests/test_evaluation.py:13]: Two tests currently failing. Will investigate

File: results/latex/bart-test-extrinsic-100.tex [hunk 3]
```diff
@@ -1,14 +1,14 @@
 \newcommand{\bartFbsOracleFactual}{64\%}
 \newcommand{\bartFbsOracleNonFactual}{36\%}
-\newcommand{\bartFbsOracleNonFactualExtrinsic}{18\%}
-\newcommand{\bartFbsOracleNonFactualIntrinsic}{13\%}
-\newcommand{\bartFbsOracleFactualExtrinsicRatio}{96\%}
+\newcommand{\bartFbsOracleNonFactualExtrinsic}{25\%}
+\newcommand{\bartFbsOracleNonFactualIntrinsic}{14\%}
+\newcommand{\bartFbsOracleFactualExtrinsicRatio}{92\%}
```
Comment (anton164) [inline results/latex/bart-test-extrinsic-100.tex:5]: Unfortunately this has a substantial impact on the ratios

File: tests/test_evaluation.py (hunk 2 above)
Comment (anton164) [reply tests/test_evaluation.py:13]: Ignoring - tests are not passing on master due to updated annotations

File: evaluate_iteration_changes.py [hunk 4]
```diff
@@ -56,12 +56,14 @@ def collect_iteration_stats(
                 current_iteration_stats["summary"][sum_id] = {
                     "summary": data["summary"]
                 }
+                failed_sums_by_id[sum_id] = iteration_idx
```
Comment (dleve123) [inline evaluate_iteration_changes.py:59]: it doesn't seem like there's anything here to specify to failed summaries..why does this work?

Comment (anton164) [reply evaluate_iteration_changes.py:59]: What do you mean? It sets the sum_id in the dict so you can just look up whether the sum_id is in the dict to determine that a summary failed

Comment (dleve123) [reply evaluate_iteration_changes.py:59]: I'm likely just missing it, but there's nothing specific to `summaries` on L51, L44 that means that all the summaries are failed, right? I presume that `summaries = iteration_data["summaries"]` includes much more than failed summaries...

**Pull request merged**